SharedPackages/BrowserServicesKit/Sources/DDGSync/SyncConnectionController.swift

```diff
@@ -86,7 +86,7 @@ public protocol SyncConnectionControlling {
     func loginAndShowDeviceConnected(recoveryKey: SyncCode.RecoveryKey, isRecovery: Bool) async throws
 }
 
-final public class SyncConnectionController: SyncConnectionControlling {
+public actor SyncConnectionController: SyncConnectionControlling {
```

Comment:
Made this an Actor so that the `isCodeHandlingInFlight` check would be safe. This is why there's so many changes. I think this is a reasonable change to make